### ISSUE TITLE
False positive with `Style/SuperArguments` cop

### ISSUE THREAD

**Issue by mwolman** [rubocop/rubocop#12937]
The `Style/SuperArguments` cop incorrectly removes arguments in the super call when the signature is identical, resulting in incorrect behavior.

--------

## Expected behavior

The cop should not remove the arguments in the super call when arguments are being passed, as it changes the behavior of the code.

### Code Example

Given the following method:
```ruby
def initialize(args = {})
  super(args)
  @models = [user]
end
``` 
Running rubocop -a results in:

```ruby
def initialize(args = {})
  super
  @models = [user]
end
``` 
This transformation is incorrect because it alters the method's behavior. Specifically, the arguments passed to initialize are no longer passed to super, which can cause unexpected issues if args contains important information.

## Steps to reproduce the problem

1. Define a method that initializes with arguments and passes those arguments to super.
2. Run rubocop -a.
3. Observe the transformation and verify that it changes the behavior of the code.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.64.0 (using Parser 3.3.1.0, rubocop-ast 1.31.3, running on ruby 3.3.1) [arm64-darwin23]
  - rubocop-capybara 2.20.0
  - rubocop-factory_bot 2.25.1
  - rubocop-performance 1.21.0
  - rubocop-rails 2.25.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.29.2
  - rubocop-rspec_rails 2.28.3
```

```
$ [bundle exec] ruby -v
ruby 3.3.1 (2024-04-23 revision c56cd86388) [arm64-darwin23]
```

**Comment by bbatsov** [comment]
> Specifically, the arguments passed to initialize are no longer passed to super, which can cause unexpected issues if args contains important information.

I'm not sure I understand what changes exactly in your example - after all if you want to pass all method args to `super` you don't need to specify them explicitly there. That's essentially the reason why this cop was introduced - to remove some redundancy. //cc @Earlopain

**Comment by mwolman** [comment]
@bbatsov Oh, sorry for the misunderstood. I'm going to close the issue. Thank you 🙂
